### PR TITLE
Hifi/bc 641 project presentation completion

### PR DESCRIPTION
**Pull request by GeoSauer**
### This PR accomplishes:

- Styles the 4 pages involved in the Project Completion flow to match the Figma HiFi. 
- Componentizes the Project Url Input and Participation Radio Inputs for reusability and proper state management.
- Refactors form submission logic to only update redux throughout the flow and only perform the database update on final for confirmation/submission.
- Adds more optional props to Primary/SecondaryButton to expand their utility.
- Adds the `trackCompletion` prop to Paginator so that it can be used as a stepper or to simply show location. 

### This PR does not accomplish:

- Styling ProjectDetails screen to match Figma HiFi.
    - The screen is close but there's definitely changes to be made that are beyond nits
    - It felt like I was getting into the weeds on this so it doesn't have the "Your team will/won't be presenting" toast and the button/modal to edit participation.
    - Since I componentized the radio input it will be pretty quick to implement in a future ticket.

### Questions/Considerations

- I know we're approaching a more serious push for componentization so just pointing out that the ProjectDetails/Presentation tab is the exact same content as the ProjectCompletion/Presentation page (minus the inputs/buttons). 
- Based on the talk I had with Dan, Erick, and Hector, it seems like the data stored in the `completedInfo` of a project can be slimmed down quite a bit.
    - currently `deployedUrl = {userID: 'www.project.com'}`, not huge but if we're only collecting one url that cannot be changed (per HiFi notes) then I propose we simply store the url in a string and leave the userID out.
    - `completedInfo` also has a `participatingMembers` prop that is an array with entries containing full user info and their decision to participate or not. It was confirmed that only one team member goes through the ProjectCompletion flow so I propose we remove `participatingMembers` and just use a prop like `presenting` that stores a boolean. This is currently implemented alongside with `participatingMembers` prop.
- Currently the pagination on the ProjectCompletion flow allows users to click ahead of the step they were one. I propose removing nav functionality from the pagination in this use-case so the user must use the buttons.
- I don't know where we're at on it yet, but we've got a TODO regarding the google survey that users can take at the end of the flow and I just wanted to check in on that. 

Edit: I'm not sure what I did to include the commits from my first ticket on this, but these changes begin on commit [9b89963](https://github.com/yeezick/bootcampr/pull/129/commits/9b899632da8b345d2d34ad10bb7b44ea7e40cb8b)

[Ticket BC-641](https://bootcamper.atlassian.net/browse/BC-641)

### CODE REVIEW

File: src/components/Inputs/ProjectUrl.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  const isUrl = string => {
+    const urlPattern = new RegExp(
+      '^(https?:\\/\\/)?' +
+        '(www\\.)?' +
+        '(([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}$',
+      'i'
+    )
+    return urlPattern.test(string)
+  }
```
Comment (yeezick) [inline src/components/Inputs/ProjectUrl.tsx:42]: export this to a util file, doesn't rely on the component at all to be utilized

File: src/screens/ProjectCompletion/PresentationPage.tsx [hunk 2]
```diff
@@ -1,182 +1,174 @@
...
-    setSelectedRadio(e.target.value)
+    //TODO: I changed the logic to rely on updating the redux through the main flow and only submitting the db update on the final submit
+    // if (!isDisabled) {
+    //   try {
+    //     setIsLoading(true)
+    //     const project = await getOneProject(projectId)
+
+    //     const member = {
+    //       user: {
+    //         _id: authUser._id,
```
Comment (yeezick) [inline src/screens/ProjectCompletion/PresentationPage.tsx:36]: if your refactoring is functional we can remove these comments

File: src/screens/ProjectCompletion/UrlPage.tsx [hunk 3]
```diff
@@ -1,125 +1,102 @@
...
+    // }
 
-      try {
-        setIsLoading(true)
-        const response = await editProject(projectID, updatedProject)
+    // const isDuplicate = Object.values(
+    //   project.completedInfo?.deployedUrl || {}
+    // ).some(url => {
+    //   const existingUrl = normalizeUrl(url)
+    //   const submittedUrl = normalizeUrl(projectUrl)
```
Comment (yeezick) [inline src/screens/ProjectCompletion/UrlPage.tsx:39]: remove whatever isn't needed

File: src/screens/ProjectCompletion/WhatsNextPage.tsx [hunk 4]
```diff
@@ -1,76 +1,82 @@
...
+  const handleCancel = () => {
+    navigate(`/project/${projectID}`)
   }
 
-  const latestMemberIndex = completedInfo.participatingMembers.length - 1
-  const latestMember = completedInfo.participatingMembers[latestMemberIndex]
-  const shouldDisplayTime = latestMember?.decision === 'Participate'
+  // const latestMemberIndex = completedInfo.participatingMembers.length - 1
+  // const latestMember = completedInfo.participatingMembers[latestMemberIndex]
+  // const userParticipating = latestMember?.decision === 'Participate'
```
Comment (yeezick) [inline src/screens/ProjectCompletion/WhatsNextPage.tsx:30]: dead code

**Review by yeezick — CHANGES_REQUESTED** [review]
implementation looks great! all i had to call out were clean up items, good job!

File: src/components/Inputs/ProjectUrl.tsx (hunk 1 above)
Comment (GeoSauer) [reply src/components/Inputs/ProjectUrl.tsx]: sounds good!

**Review by yeezick — APPROVED** [review]

**Pull request merged**